### PR TITLE
py-s3transfer: update to 0.1.12

### PR DESCRIPTION
**Pull request by emcrisostomo**
###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix
- [x] update

###### Tested on

macOS 10.13.2
Xcode 9.2

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

Fixes: https://trac.macports.org/ticket/55526

### CODE REVIEW

File: python/py-s3transfer/Portfile [hunk 1]
```diff
@@ -15,9 +15,8 @@ long_description    ${description}
 homepage            https://github.com/boto/${python.rootname}
 master_sites        pypi:s/${python.rootname}
 distname            ${python.rootname}-${version}
-checksums           md5     976734d3ec36b8fc2c1b505a907c3259 \
```
Comment (l2dy) [inline python/py-s3transfer/Portfile]: Why did you remove the md5 checksum?

Comment (emcrisostomo) [reply python/py-s3transfer/Portfile]: Since all other ports I've worked with had `rmd160` and `sha256` I was under the impression that it was unnecessary to include it as well. Even the MacPorts documentation (section 4.2) use `rmd160` and `sha256` when exemplifying the construction of a new port.

Comment (mf2k) [reply python/py-s3transfer/Portfile]: Yes, `md5` is considered obsolete. However, it can be included if upstream publishes it for verification.

**Comment by mf2k** [comment]
Please modify the comment to say:
Fixes: https://trac.macports.org/ticket/55526

**Comment by emcrisostomo** [comment]
@mf2k Done

File: python/py-s3transfer/Portfile (hunk 1 above)
Comment (emcrisostomo) [reply python/py-s3transfer/Portfile]: Thanks, ok, I won't drop them any longer.

**Comment by l2dy** [comment]
>Please modify the comment to say:
>Fixes: https://trac.macports.org/ticket/55526

Not the comment, this must go into the commit message.

File: python/py-s3transfer/Portfile (hunk 1 above)
Comment (l2dy) [reply python/py-s3transfer/Portfile]: Please keep it here too.

Comment (emcrisostomo) [reply python/py-s3transfer/Portfile]: Restored.

**Comment by emcrisostomo** [comment]
> Not the comment, this must go into the commit message.

Done

**Comment by l2dy** [comment]
Please add "Fixes: ..." to the commit message body, see https://trac.macports.org/wiki/CommitMessages#guidelines.
